### ISSUE TITLE
ElixirLS crashes on windows when home folder contains opening parentheses "("

### ISSUE THREAD

**Issue by sejnub** [elixir-lsp/elixir-ls#756]
### Environment

* Elixir & Erlang versions (elixir --version): erlang 25.0.4; elixir 1.14.0-otp-25

* Elixir Language Server version: 
* Operating system: Windows 10
* Editor or IDE name (e.g. Emacs/VSCode): VSCode
* Editor Plugin/LSP Client name and version: ElixirLS: Elixir support and debugger v0.11.0

### Current behavior

I create an elixir project. I then open my first elixir source file.
Then ElixirLS crashes 5 time with the following log

```log
Der Befehl "c:\Users\HeinerBunjes" ist entweder falsch geschrieben oder
konnte nicht gefunden werden.
[Info  - 12:29:55] Connection to server got closed. Server will restart.
Der Befehl "c:\Users\HeinerBunjes" ist entweder falsch geschrieben oder
konnte nicht gefunden werden.
[Info  - 12:29:55] Connection to server got closed. Server will restart.
Der Befehl "c:\Users\HeinerBunjes" ist entweder falsch geschrieben oder
konnte nicht gefunden werden.
[Info  - 12:29:55] Connection to server got closed. Server will restart.
Der Befehl "c:\Users\HeinerBunjes" ist entweder falsch geschrieben oder
konnte nicht gefunden werden.
[Info  - 12:29:55] Connection to server got closed. Server will restart.
Der Befehl "c:\Users\HeinerBunjes" ist entweder falsch geschrieben oder
konnte nicht gefunden werden.
[Error - 12:29:55] Connection to server got closed. Server will not be restarted.
```

My guess is that the stupid opening parenthesis after `HeinerBunjes` that windows put into my user name is causing the problem. The user is an Azure AD user.

I created another user on the same windows machine that has a straight user name without special characters. With this straight user ElixirLS works fine and does not crash.

When I set trace to verbose it says:

```log
[Trace - 13:45:27] Sending request 'initialize - (0)'.
Params: {
    "processId": 14488,
    "clientInfo": {
        "name": "Visual Studio Code",
        "version": "1.72.2"
    },
    "locale": "de",
    "rootPath": "c:\\Users\\HeinerBunjes(beenera\\OneDrive - BEENIC GmbH\\regime\\work\\commp",
    "rootUri": "file:///c%3A/Users/HeinerBunjes%28beenera/OneDrive%20-%20BEENIC%20GmbH/regime/work/commp",
    "capabilities": {
        "workspace": {
            "applyEdit": true,
            "workspaceEdit": {
                "documentChanges": true,
                "resourceOperations": [
                    "create",
                    "rename",
                    "delete"
                ],
                "failureHandling": "textOnlyTransactional",
                "normalizesLineEndings": true,
                "changeAnnotationSupport": {
                    "groupsOnLabel": true
                }
            },
            "didChangeConfiguration": {
                "dynamicRegistration": true
            },
            "didChangeWatchedFiles": {
                "dynamicRegistration": true
            },
            "symbol": {
                "dynamicRegistration": true,
                "symbolKind": {
                    "valueSet": [
                        1,
                        2,
                        3,
                        4,
                        5,
                        6,
                        7,
                        8,
                        9,
                        10,
                        11,
                        12,
                        13,
                        14,
                        15,
                        16,
                        17,
                        18,
                        19,
                        20,
                        21,
                        22,
                        23,
                        24,
                        25,
                        26
                    ]
                },
                "tagSupport": {
                    "valueSet": [
                        1
                    ]
                }
            },
            "codeLens": {
                "refreshSupport": true
            },
            "executeCommand": {
                "dynamicRegistration": true
            },
            "configuration": true,
            "workspaceFolders": true,
            "semanticTokens": {
                "refreshSupport": true
            },
            "fileOperations": {
                "dynamicRegistration": true,
                "didCreate": true,
                "didRename": true,
                "didDelete": true,
                "willCreate": true,
                "willRename": true,
                "willDelete": true
            }
        },
        "textDocument": {
            "publishDiagnostics": {
                "relatedInformation": true,
                "versionSupport": false,
                "tagSupport": {
                    "valueSet": [
                        1,
                        2
                    ]
                },
                "codeDescriptionSupport": true,
                "dataSupport": true
            },
            "synchronization": {
                "dynamicRegistration": true,
                "willSave": true,
                "willSaveWaitUntil": true,
                "didSave": true
            },
            "completion": {
                "dynamicRegistration": true,
                "contextSupport": true,
                "completionItem": {
                    "snippetSupport": true,
                    "commitCharactersSupport": true,
                    "documentationFormat": [
                        "markdown",
                        "plaintext"
                    ],
                    "deprecatedSupport": true,
                    "preselectSupport": true,
                    "tagSupport": {
                        "valueSet": [
                            1
                        ]
                    },
                    "insertReplaceSupport": true,
                    "resolveSupport": {
                        "properties": [
                            "documentation",
                            "detail",
                            "additionalTextEdits"
                        ]
                    },
                    "insertTextModeSupport": {
                        "valueSet": [
                            1,
                            2
                        ]
                    }
                },
                "completionItemKind": {
                    "valueSet": [
                        1,
                        2,
                        3,
                        4,
                        5,
                        6,
                        7,
                        8,
                        9,
                        10,
                        11,
                        12,
                        13,
                        14,
                        15,
                        16,
                        17,
                        18,
                        19,
                        20,
                        21,
                        22,
                        23,
                        24,
                        25
                    ]
                }
            },
            "hover": {
                "dynamicRegistration": true,
                "contentFormat": [
                    "markdown",
                    "plaintext"
                ]
            },
            "signatureHelp": {
                "dynamicRegistration": true,
                "signatureInformation": {
                    "documentationFormat": [
                        "markdown",
                        "plaintext"
                    ],
                    "parameterInformation": {
                        "labelOffsetSupport": true
                    },
                    "activeParameterSupport": true
                },
                "contextSupport": true
            },
            "definition": {
                "dynamicRegistration": true,
                "linkSupport": true
            },
            "references": {
                "dynamicRegistration": true
            },
            "documentHighlight": {
                "dynamicRegistration": true
            },
            "documentSymbol": {
                "dynamicRegistration": true,
                "symbolKind": {
                    "valueSet": [
                        1,
                        2,
                        3,
                        4,
                        5,
                        6,
                        7,
                        8,
                        9,
                        10,
                        11,
                        12,
                        13,
                        14,
                        15,
                        16,
                        17,
                        18,
                        19,
                        20,
                        21,
                        22,
                        23,
                        24,
                        25,
                        26
                    ]
                },
                "hierarchicalDocumentSymbolSupport": true,
                "tagSupport": {
                    "valueSet": [
                        1
                    ]
                },
                "labelSupport": true
            },
            "codeAction": {
                "dynamicRegistration": true,
                "isPreferredSupport": true,
                "disabledSupport": true,
                "dataSupport": true,
                "resolveSupport": {
                    "properties": [
                        "edit"
                    ]
                },
                "codeActionLiteralSupport": {
                    "codeActionKind": {
                        "valueSet": [
                            "",
                            "quickfix",
                            "refactor",
                            "refactor.extract",
                            "refactor.inline",
                            "refactor.rewrite",
                            "source",
                            "source.organizeImports"
                        ]
                    }
                },
                "honorsChangeAnnotations": false
            },
            "codeLens": {
                "dynamicRegistration": true
            },
            "formatting": {
                "dynamicRegistration": true
            },
            "rangeFormatting": {
                "dynamicRegistration": true
            },
            "onTypeFormatting": {
                "dynamicRegistration": true
            },
            "rename": {
                "dynamicRegistration": true,
                "prepareSupport": true,
                "prepareSupportDefaultBehavior": 1,
                "honorsChangeAnnotations": true
            },
            "documentLink": {
                "dynamicRegistration": true,
                "tooltipSupport": true
            },
            "typeDefinition": {
                "dynamicRegistration": true,
                "linkSupport": true
            },
            "implementation": {
                "dynamicRegistration": true,
                "linkSupport": true
            },
            "colorProvider": {
                "dynamicRegistration": true
            },
            "foldingRange": {
                "dynamicRegistration": true,
                "rangeLimit": 5000,
                "lineFoldingOnly": true
            },
            "declaration": {
                "dynamicRegistration": true,
                "linkSupport": true
            },
            "selectionRange": {
                "dynamicRegistration": true
            },
            "callHierarchy": {
                "dynamicRegistration": true
            },
            "semanticTokens": {
                "dynamicRegistration": true,
                "tokenTypes": [
                    "namespace",
                    "type",
                    "class",
                    "enum",
                    "interface",
                    "struct",
                    "typeParameter",
                    "parameter",
                    "variable",
                    "property",
                    "enumMember",
                    "event",
                    "function",
                    "method",
                    "macro",
                    "keyword",
                    "modifier",
                    "comment",
                    "string",
                    "number",
                    "regexp",
                    "operator"
                ],
                "tokenModifiers": [
                    "declaration",
                    "definition",
                    "readonly",
                    "static",
                    "deprecated",
                    "abstract",
                    "async",
                    "modification",
                    "documentation",
                    "defaultLibrary"
                ],
                "formats": [
                    "relative"
                ],
                "requests": {
                    "range": true,
                    "full": {
                        "delta": true
                    }
                },
                "multilineTokenSupport": false,
                "overlappingTokenSupport": false
            },
            "linkedEditingRange": {
                "dynamicRegistration": true
            }
        },
        "window": {
            "showMessage": {
                "messageActionItem": {
                    "additionalPropertiesSupport": true
                }
            },
            "showDocument": {
                "support": true
            },
            "workDoneProgress": true
        },
        "general": {
            "regularExpressions": {
                "engine": "ECMAScript",
                "version": "ES2020"
            },
            "markdown": {
                "parser": "marked",
                "version": "1.1.0"
            }
        }
    },
    "trace": "verbose",
    "workspaceFolders": [
        {
            "uri": "file:///c%3A/Users/HeinerBunjes%28beenera/OneDrive%20-%20BEENIC%20GmbH/regime/work/commp",
            "name": "commp"
        }
    ]
}


Der Befehl "c:\Users\HeinerBunjes" ist entweder falsch geschrieben oder
konnte nicht gefunden werden.
[Info  - 13:45:28] Connection to server got closed. Server will restart.
[Trace - 13:45:28] Sending request 'initialize - (0)'.
Params: ~~~the same params again; I deleted them here~~~



[Error - 13:45:28] Connection to server is erroring. Shutting down server.
[Trace - 13:45:28] Sending request 'shutdown - (1)'.
Der Befehl "c:\Users\HeinerBunjes" ist entweder falsch geschrieben oder
konnte nicht gefunden werden.

```

What can I do to prevent ElixirLS from crashing?

**Comment by lukaszsamson** [comment]
I cannot reproduce this on v0.12 and windows 11. I created a windows directory structure `c:\Users\myuser\HeinerBunjes(beenera\OneDrive - BEENIC GmbH\regime\work\commp\mytestproj`. I also tried putting it in One Drive dir. In both cases it works as expected.
Please create a github repository that reliably reproduces this crash

**Comment by lukaszsamson** [comment]
I suspected that there may be a problem in the way we convert correctly path to URI
```
"rootPath": "c:\\Users\\HeinerBunjes(beenera\\OneDrive - BEENIC GmbH\\regime\\work\\commp",
"rootUri": "file:///c%3A/Users/HeinerBunjes%28beenera/OneDrive%20-%20BEENIC%20GmbH/regime/work/commp",
```
but it works correctly. There were some changes in the startup code dealing with project dir since the issue was initially reported. Maybe one of them fixed it. Since there is no reproduction steps I'm going to close. Feel free to reopen if you can provide a failing test or repo that reproduces it.

**Comment by lukaszsamson** [comment]
Another possible fix for the underlaying issue landed in v0.14.4
